### PR TITLE
[REVIEW] Remove SNMG code

### LINKED ISSUE #665: [BUG] Remove gdf_column in SNMG features
degree, coo2csr, PageRank

### PR DESCRIPTION
**Pull request by ChuckHastings**
Remove references to snmg code, it will be replaced by OPG.

closes #665 

The SNMG code uses the old gdf_column and old RMM calls.  The plan is to replace this capability with OPG in the next few weeks.  Rather than spend time porting the SNMG code to use the new C++ graph objects and the RAII RMM constructs, we are disabling this feature until the OPG code is ready.

I merely commented things out for now.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by ChuckHastings** [comment]
> * We probably want to remove it from the C++ header (or raise a c++ exception): https://github.com/rapidsai/cugraph/blob/9df98992efc2a8c2db62638ced08c9f8ad9af5e9/cpp/include/algorithms.h#L99
> * Should we update something so that cython does not build smng and dask features?

Missed the algorithms.h change... did that.

Deleted the cython bindings (not really relevant for OPG anyway, the API will be totally different).

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -349,16 +349,16 @@ add_library(cugraph SHARED
...
-    src/snmg/utils.cu
+    #src/snmg/utils.cu
     src/components/connectivity.cu
     src/centrality/katz_centrality.cu
     src/centrality/betweenness_centrality.cu
-    src/snmg/degree/degree.cu
-    src/snmg/COO2CSR/COO2CSR.cu
+    #src/snmg/degree/degree.cu
+    #src/snmg/COO2CSR/COO2CSR.cu
     src/nvgraph/arnoldi.cu
```
Comment (seunghwak) [inline cpp/CMakeLists.txt:362]: Can we just delete these instead of commenting out? We can see github history when we need to get these back.

**Comment by BradReesWork** [comment]
rerun tests

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (BradReesWork) [reply cpp/CMakeLists.txt:357]: They should be removed since we are not going bad to SNMG.  OPG is the future :-)

Comment (ChuckHastings) [reply cpp/CMakeLists.txt:357]: (already did that)

**Comment by BradReesWork** [comment]
rerun tests

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -174,42 +174,42 @@ endif(BUILD_TESTS)
...
+#
+#add_library(NVCategory SHARED IMPORTED ${NVCATEGORY_LIBRARY})
+#if (NVSTRINGS_INCLUDE AND NVCATEGORY_LIBRARY)
+#    set_target_properties(NVCategory PROPERTIES IMPORTED_LOCATION ${NVCATEGORY_LIBRARY})
+#endif (NVSTRINGS_INCLUDE AND NVCATEGORY_LIBRARY)
+#
+#add_library(NVText SHARED IMPORTED ${NVTEXT_LIBRARY})
+#if (NVSTRINGS_INCLUDE AND NVTEXT_LIBRARY)
+#    set_target_properties(NVText PROPERTIES IMPORTED_LOCATION ${NVTEXT_LIBRARY})
+#endif (NVSTRINGS_INCLUDE AND NVTEXT_LIBRARY)
```
Comment (seunghwak) [inline cpp/CMakeLists.txt:212]: My last comment on this PR, can you delete this as well? (instead of just commenting out). Besides this, LGTM.

**Comment by BradReesWork** [comment]
rerun tests

**Pull request merged**